### DISCUSSION TITLE
Consulta de enunciado - QMP segunda iteración

### DISCUSSION THREAD

**Discussion by AlejandroDeheza** [dds-jv/foro#207] [Consultas de objetos] [answered]
Buenas! estoy tratando de resolver el ejercicio pero no estoy entendiendo bien el siguiente requerimiento:

~~~
Como usuarie de QuéMePongo, quiero crear una prenda especificando en segundo lugar los aspectos relacionados a su 
material (colores, material, trama, etc) para evitar elegir materiales inconsistentes con el tipo de prenda.
~~~

¿Qué materiales serian inconsistentes según el tipo de prenda? ¿En qué me debería basar para saberlo en todos los casos?

No se si parte del ejercicio es tratar de pensar esto uno mismo, lo pregunto por las dudas.
Saludos!

**Comment by nanderson87** [comment, marked as answer]
Buenas:

Lo importante en este caso es poder contemplar la restricción y tenerla en cuenta al momento de modelar.  Es decir entender que exista esta noción de que ciertos materiales no van a tener sentido en ciertos tipos de prenda y que de algún modo lo vamos a querer controlar. Ahora bien, cuáles son esas relaciones? las que vos prefieras ya que son solo a modo ilustrativo.

Por ejemplo: Si tengo el material goma. Por ahí tiene sentido poder tener  ojotas de goma pero no tanto que puedas cargar en el sistema un buzo de goma.

Otro ejemplo: Supongamos que en en la primera iteración tenías el material lana. Y queres decir que las medias, las gorras y camperas pueden ser de lana. Pero los pantalones no, tmb va a estar bien. (aunque en el mundo real puede o no existir pantalones de lana).

En resumen si es importante contemplar la validación en el modelado pero no tiene tanta importancia que se va a validar en detalle. Es más,  seguramente en esta entrega, este punto de validación por ahí tiene menos relevancia que otras problematicas más interesantes que nos plantea el ejercicio como puede ser poder tener un borrador o poder continuar la carga de una prenda después.

Espero haber podido responder tu consulta. Cualquier cosas nos avisas.

**Reply by AlejandroDeheza** [reply]
Buenísimo, muchas gracias!
